app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/NiaApp.kt

```diff
@@ -162,6 +162,7 @@ fun NiaApp(appState: NiaAppState) {
                     Column(Modifier.fillMaxSize()) {
                         // Show the top app bar on top level destinations.
                         val destination = appState.currentTopLevelDestination
+                        val shouldShowTopAppBar = destination != null
                         if (destination != null) {
```

Comment:
```suggestion
                        if (shouldShowTopAppBar) {
```
This is same logic from ```shouldShowTopAppBar```

Reply:
Null=$Sgzeus23

Reply:
This is the same logic, but using `shouldShowTopAppBar` directly results in not being able to smart-cast `destination` to be non-nullable below.

Reply:
Good, I missed the point of ```smart cast```.